nc/source/physics/PhysicsSystem.cpp

```diff
@@ -7,21 +8,90 @@
 #include "job/JobSystem.h"
 #include "debug/Profiler.h"
 
+namespace
+{
+    nc::physics::PhysicsSystem* g_physicsSystem = nullptr;
+}
+
 namespace nc::physics
 {
+    void AddJoint(Entity entityA, Entity entityB, const Vector3& anchorA, const Vector3& anchorB, float bias, float softness)
+    {
+        g_physicsSystem->AddJoint(entityA, entityB, anchorA, anchorB, bias, softness);
+    }
+
+    void RemoveJoint(Entity entityA, Entity entityB)
+    {
+        g_physicsSystem->RemoveJoint(entityA, entityB);
+    }
+
+    void RemoveAllJoints(Entity entity)
+    {
+        g_physicsSystem->RemoveAllJoints(entity);
+    }
+
     #ifdef USE_VULKAN
     PhysicsSystem::PhysicsSystem(graphics::Graphics2* graphics, job::JobSystem* jobSystem)
     #else
     PhysicsSystem::PhysicsSystem(graphics::Graphics* graphics, job::JobSystem* jobSystem)
     #endif
         : m_cache{},
+          m_joints{},
           m_clickableSystem{graphics},
           m_hullColliderManager{},
           m_jobSystem{jobSystem}
           #ifdef NC_DEBUG_RENDERING
           , m_debugRenderer{graphics}
           #endif
     {
+        g_physicsSystem = this;
+    }
+
+    void PhysicsSystem::AddJoint(Entity entityA, Entity entityB, const Vector3& anchorA, const Vector3& anchorB, float bias, float softness)
+    {
+        m_joints.emplace_back
+        (
+            entityA,
+            entityB,
+            nullptr,
+            nullptr,
+            DirectX::XMLoadVector3(&anchorA),
+            DirectX::XMLoadVector3(&anchorB),
+            DirectX::XMMATRIX{},
+            DirectX::XMVECTOR{},
+            DirectX::XMVECTOR{},
+            DirectX::XMVECTOR{},
+            DirectX::XMVECTOR{},
+            bias,
+            softness
+        );
+    }
+
+    void PhysicsSystem::RemoveJoint(Entity entityA, Entity entityB)
```

Comment:
In Physics.h, you say that joints must be manually removed if an entity is destroyed. Wont that not work here? (If Entity entityB no longer exists, for example)